src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -277,6 +281,8 @@ protected Object executeObject(ExecutionContext executionContext, ExecutionStrat
 
     private BiConsumer<List<Object>, Throwable> buildFieldValueMap(List<String> fieldNames, CompletableFuture<Map<String, Object>> overallResult, ExecutionContext executionContext) {
         return (List<Object> results, Throwable exception) -> {
+            exception = executionContext.possibleCancellation(exception);
```

Comment:
This does a trick - if the Throwable is already set - leave it set - otherwise check if we are cancelled and make the throwable be that